### PR TITLE
Added `onTaskComplete` callback option

### PR DESCRIPTION
**Pull request by tmeasday**
Telescoping on https://github.com/chromaui/chromatic-cli/pull/755

And ability to pass options to the node API.

This will allow us to get access to the context at the completion of each task, in particular to get the build id at the end of the `announce` task.

### CODE REVIEW

File: bin-src/main.ts [hunk 1]
```diff
@@ -77,12 +77,12 @@ export async function runAll(ctx) {
...
 }
 
-export async function runBuild(ctx: Context) {
+export async function runBuild(ctx: Context, extraOptions?: Partial<Options>) {
   ctx.log.info('');
   ctx.log.info(intro(ctx));
 
   try {
-    ctx.options = await getOptions(ctx);
+    ctx.options = await getOptions(ctx, extraOptions);
```
Comment (ghengeveld) [inline bin-src/main.ts:85]:
```suggestion
    ctx.options = await getOptions(ctx);
    if (extraOptions) ctx.options = { ...ctx.options, ...extraOptions };
```
This makes it more apparent that "all bets are off" at this point.

File: bin-src/lib/tasks.ts [hunk 2]
```diff
@@ -15,6 +15,8 @@ export const createTask = ({ title, steps, ...config }): Listr.ListrTask<Context
       // eslint-disable-next-line no-await-in-loop
       await step(ctx, task);
     }
+
+    ctx.options.onTaskComplete?.(ctx);
```
Comment (ghengeveld) [inline bin-src/lib/tasks.ts:19]:
```suggestion
    ctx.options.onTaskComplete?.({ ...ctx });
```
This should ensure `ctx.title` still refers to the title of the current (finished) task rather than the next one.

File: node-src/index.ts [hunk 3]
```diff
@@ -24,7 +24,13 @@ interface Output {
...
 }
 
-export async function runChromatic(flags: Flags): Promise<Output> {
+export async function runChromaticFull({
+  flags = {},
+  options = {},
+}: {
+  flags?: Flags;
+  options?: Partial<Options>;
+}): Promise<Output> {
```
Comment (ghengeveld) [inline node-src/index.ts:33]: I'm not digging the name `runChromaticFull`. What does "full" mean in this context? I think we should treat this node entry point as a public API for anyone who wants to integrate more deeply with our CLI. I think it should either be a descriptive and truthful name, or be something quite generic like `run` or `main`.

**Review by ghengeveld — CHANGES_REQUESTED** [review]

File: bin-src/lib/getOptions.ts [hunk 4]
```diff
@@ -75,6 +78,7 @@ export default function getOptions({ argv, env, flags, log, packageJson }: Conte
     branchName,
     patchHeadRef,
     patchBaseRef,
+    ...extraOptions,
```
[the remark this answers is not in the document]
Comment (tmeasday) [reply bin-src/lib/getOptions.ts]: @ghengeveld the main issue with your approach is then the checks below (e.g. for missing project token) start hassling me if I pass the token in via options (rather than flags). We should have a chat about how we want this to work.

File: action-src/main.ts [hunk 5]
```diff
@@ -109,34 +109,35 @@ async function run() {
...
-      projectToken,
-      repositorySlug: maybe(repositorySlug),
-      skip: maybe(skip),
-      storybookBaseDir: maybe(storybookBaseDir),
-      storybookBuildDir: maybe(storybookBuildDir),
-      storybookConfigDir: maybe(storybookConfigDir),
-      traceChanged: maybe(traceChanged),
-      untraced: maybe(untraced),
-      zip: maybe(zip, false),
-      junitReport: maybe(junitReport, false),
```
Comment (ghengeveld) [inline action-src/main.ts:139]: Was this removal intentional?

**Review by ghengeveld — APPROVED** [review]

File: action-src/main.ts (hunk 5 above)
Comment (tmeasday) [reply action-src/main.ts:139]: It's just moved to the flags option
